tests/unit/mb_enhanced_cover_art_uploads/providers/tidal.test.ts

```diff
@@ -85,6 +85,8 @@ describe('tidal provider', () => {
         const extractionFailedCases = [{
             desc: 'non-existent release',
             url: 'https://listen.tidal.com/album/1',
+            // FIXME: Tidal provider doesn't use the base `fetchPage` and only throws a generic error.
```

Comment:
Is this a non-trivial fix or do you simply haven't looked at it yet?

Reply:
It's easy enough to fix by wrapping the custom request in a try-catch, but that'd be another case of
```
let response;
try {
    response = await ...
} catch ...
```
and that pattern is starting to annoy me…

What I'd like to do instead is add some sort of optional "error factory" to `gmxhr` to create custom errors based on some conditions but that's a less-trivial change. Moreover, I've got some other ideas in the pipeline (logging responses somewhere so that they can be included in a bug report to debug e.g. #521), and for that I was thinking of having a unified interface for `gmxhr` and `fetch` so that both can be logged (not essential, but it'd be nice to be able to use them in the same way). So IMO it'd make sense to group those changes together. TBH it'd make sense to group those changes into this PR as well.

Reply:
The error message issue has been addressed, the interface unification is something I'm still considering and will go into the PR with improved logging if I decide that the unification is necessary.